### PR TITLE
docs(net): add example docs and fix links

### PR DESCRIPTION
**Pull request by mattsse**
- Adds high level example doc
- additional convenience in `config`
- fix broken intra doc links

### CODE REVIEW

File: crates/net/network/src/config.rs [hunk 1]
```diff
@@ -12,6 +14,19 @@ use std::{
     sync::Arc,
 };
 
+/// reexports for convenience
+#[doc(hidden)]
```
Comment (gakonst) [inline crates/net/network/src/config.rs:18]: Is this a common technique? How come you didn't pub use each thing?

File: crates/net/network/src/lib.rs [hunk 2]
```diff
@@ -12,9 +13,44 @@
 //! In order for a node to join the ethereum p2p network it needs to know what nodes are already
 //! port of that network. This includes public identities (public key) and addresses (where to reach
 //! them).
+//!
+//! ## Usage
+//!
+//! ### Configure and launch the network
```
Comment (gakonst) [inline crates/net/network/src/lib.rs:19]: Beautiful. And we'll add docs later on how to use the network, how to instantiate the block provider etc

**Review by gakonst — APPROVED** [review]
Lovely

File: crates/net/network/src/config.rs (hunk 1 above)
Comment (mattsse) [reply crates/net/network/src/config.rs:18]: can't remember where I picked this up, but imo this is cleaner for external reexports and highlights that this should indeed be pub use

**Pull request merged**